frontend/src/components/ProjectStudentComponent.tsx

```diff
@@ -116,30 +153,44 @@ export default function ProjectStudentComponent(props: { project: ProjectStudent
                                 </table>
                             </div>
                         </div>
+                        {error &&
+                            <div className="notification is-danger" style={{width: "75%"}}>
+                                {error}
+                            </div>
+                        }
+                        {success &&
+                            <div className="notification is-success" style={{width: "75%"}}>
+                                {success}
+                            </div>
+                        }
                         <div className="field is-horizontal">
                             <div className="field-label">
                                 <label className="label">{"> " + t('project.submission.tag') + ":"}</label>
                             </div>
+
                             <div className="field-body">
                                 <ul className="field"> {/* Deze moet blijven */}
                                     <div className="submission-file-download-upload">
                                         {props.project.submission != null &&
                                             <div className={"submission-file-download mb-3"}>
                                                 {newSelectedFile
-                                                    ? <label className={"mr-3 highlight"}>{selectedFileName}</label>
-                                                    : <label className={"mr-3"}>{selectedFileName}</label>
+                                                    ? <label className={"mr-3 highlight"}>{file?.name}</label>
+                                                    : <label className={"mr-3"}>{file?.name}</label>
                                                 }
-                                                <button className="button">
+                                                <button className="button" onClick={() => void downloadLatestSubmission()}>
                                                     <FaDownload/>
                                                 </button>
                                             </div>
                                         }
+
                                         <div id="file-js" className="field is-horizontal">
                                             <label className="file-label">
-                                                <input className="file-input" type="file" name="resume" onChange={e => handleFileChange(e.target.files)}/>
+                                                <input className="file-input" type="file" name="resume"
+                                                       onChange={e => selectFile(e)}/>
```

Comment:
Ik vermoed dat je hier gewoon de functie kan meegeven ipv dit in een lambda te wrappen.